library/ssl_msg.c

```diff
@@ -3179,7 +3180,8 @@ int mbedtls_ssl_write_record( mbedtls_ssl_context *ssl, uint8_t force_flush )
         ( ret = mbedtls_ssl_flush_output( ssl ) ) != 0 )
     {
         MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_ssl_flush_output", ret );
-        return( ret );
+        if(ret != MBEDTLS_ERR_SSL_WANT_WRITE)
```

Comment:
same here, we may need propagate MBEDTLS_ERR_SSL_WANT_WRITE to upper level

Reply:
Agree